src/rdb/rdb_raft.c

```diff
@@ -1719,7 +1729,13 @@ rdb_raft_check_state(struct rdb *db, const struct rdb_raft_state *state,
 		rc = compaction_rc;
 	switch (rc) {
 	case -DER_NOMEM:
+	case -DER_NOSPACE:
 		if (leader) {
+			uint64_t cur_idx = raft_get_current_idx(db->d_raft);
+
+			/* No space / desperation: compact to current index */
+			(void) rdb_raft_compact_to_index(db, cur_idx);
```

Comment:
In daos we don't use this "(void)" notation. :) No need to refresh this PR though.